packages/devtools_app/lib/src/debugger/debugger_screen.dart

```diff
@@ -234,6 +240,51 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody>
       },
     );
   }
+
+  Widget _callStackRightChild() {
+    return CopyToClipboardControl(
+      dataProvider: () =>
+          controller.stackFramesWithLocation.value
+              .map((frame) {
+                final asyncMarker =
+                    frame.frame.kind == FrameKind.kAsyncSuspensionMarker;
+                return '${_descriptionFor(frame)}${asyncMarker ? null : ' ' + _locationFor(frame)}';
+              })
+              .toList()
+              .join('\n') ??
+          '',
+      buttonKey: DebuggerScreenBody.copyToClipboardButtonKey,
+    );
+  }
+
+  String _descriptionFor(StackFrameAndSourcePosition frame) {
```

Comment:
these helpers are already defined in call_stack.dart. In my comment above I suggested moving these to the `StackFrameAndSourcePosition` class. Once you do that, please remove them from here as well as call_stack.dart.